File: juke/ios/SpotifyBridge.m
//
//  SpotifyBridge.m
//  juke
//
//  Created by Harry Verhoef on 06/08/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//
#import "SpotifyBridge.h"
#import "SpotifyiOS.framework/Headers/SpotifyiOS.h"
@implementation SpotifyBridge

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(
  sessionManager:(SPTSessionManager *)manager didInitiateSession:(SPTSession *)session {
    NSLog(@"success: %@", session);
});

RCT_EXPORT_METHOD(
  sessionManager:(SPTSessionManager *)manager didFailWithError:(NSError *)error {
    NSLog(@"fail: %@", error);
});

RCT_EXPORT_METHOD(
  sessionManager:(SPTSessionManager *)manager didRenewSession:(SPTSession *)session {
    NSLog(@"renewed: %@", session);
});

+ SString *spotifyClientID = @"ff19e2ea3546447e916e43dcda51a298";
+ NSURL *spotifyRedirectURL = [NSURL URLWithString:@"http://harrys-macbook-pro.local:3000/spotify-login-callback"];

+ self.configuration  = [[SPTConfiguration alloc] initWithClientID:spotifyClientID redirectURL:spotifyRedirectURL];

@end

File: juke/ios/SpotifyBridge.swift
//
//  SpotifyBridge.swift
//  juke
//
//  Created by Harry Verhoef on 09/08/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation
import
@objc(SpotifyBridge)
class SpotifyBrudge: NSObject {
  @objc
  func constantsToExport() -> [AnyHashable : Any]! {
    return ["initialCount": 0]
  }
  
  func sessionManager(manager: SPTSessionManager, didInitiate session: SPTSession) {
    print("success", session)
  }
  func sessionManager(manager: SPTSessionManager, didFailWith error: Error) {
    print("fail", error)
  }
  func sessionManager(manager: SPTSessionManager, didRenew session: SPTSession) {
    print("renewed", session)
  }
}
